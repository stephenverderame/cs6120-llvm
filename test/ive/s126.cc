// NOLINTBEGIN
#include <cstdio>
const int LEN2 = 100;
const int ntimes = 1000;
int main()
{
    int k;
    for (int nl = 0; nl < 10 * (ntimes / LEN2); nl++) {
        k = 1;
        for (int i = 0; i < LEN2; i++) {
            for (int j = 1; j < LEN2; j++) {
                printf("%d %d\n", j - 1, k - 1);
                ++k;
            }
            ++k;
        }
    }
}
// NOLINTEND