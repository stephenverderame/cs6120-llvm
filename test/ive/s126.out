Loop: 
9:                                                ; preds = %17, %8
  %.1 = phi i32 [ %.01, %8 ], [ %16, %17 ]
  %.0 = phi i32 [ 1, %8 ], [ %18, %17 ]
  %10 = icmp slt i32 %.0, 100
  br i1 %10, label %12, label %11

Derived IV: %13  = .0 - i32 1
Derived IV: %14  = .1 - i32 1
Basic IV:   %16 = add nsw i32 %.1, 1
Basic IV:   %18 = add nsw i32 %.0, 1
Basic IV:   %.0 = phi i32 [ 1, %8 ], [ %18, %17 ]
Basic IV:   %.1 = phi i32 [ %.01, %8 ], [ %16, %17 ]
Loop: 
5:                                                ; preds = %25, %4
  %.03 = phi i32 [ 0, %4 ], [ %26, %25 ]
  %.01 = phi i32 [ 1, %4 ], [ %24, %25 ]
  %6 = icmp slt i32 %.03, 100
  br i1 %6, label %8, label %7

Basic IV:   %10 = phi i32 [ -1, %8 ], [ %21, %20 ]
Basic IV:   %11 = phi i32 [ -1, %8 ], [ %18, %20 ]
Basic IV:   %18 = add i32 %11, 1
Basic IV:   %21 = add i32 %10, 1
Basic IV:   %26 = add nsw i32 %.03, 1
Basic IV:   %.03 = phi i32 [ 0, %4 ], [ %26, %25 ]
Loop: 
1:                                                ; preds = %28, %0
  %.02 = phi i32 [ 0, %0 ], [ %29, %28 ]
  %2 = icmp slt i32 %.02, 100
  br i1 %2, label %4, label %3

Basic IV:   %29 = add nsw i32 %.02, 1
Basic IV:   %.02 = phi i32 [ 0, %0 ], [ %29, %28 ]
